src/frontend/src/btc/schedulers/btc-wallet.scheduler.ts

```diff
@@ -45,16 +66,28 @@ export class BtcWalletScheduler implements Scheduler<PostMessageDataRequestBtc>
 			network: bitcoinNetwork
 		});
 
-		// TODO: Fetch and parse transactions
-		const transactions: BitcoinTransaction[] = [];
+		let uncertifiedTransactions: CertifiedData<BitcoinTransaction>[] = [];
+
+		if (data?.shouldFetchTransactions) {
```

Comment:
Maybe we can extract a function for this? Like a private method?

Reply:
Done. Also did the same for loading balance.